### DISCUSSION TITLE
JPH_EXTERNAL_PROFILE dynamic linking

### DISCUSSION THREAD

**Discussion by tanner00** [jrouwe/JoltPhysics#1163] [Ideas]
I build and use Jolt as a dynamic library (outside of my build) and I'd like to hook it up to an external profiler. Currently, `JPH::ExternalProfileMeasurement` assumes static linkage. Similar to `JPH::TraceFunction` and `JPH::AssertFailedFunction`, handlers could be installed by the client application at runtime to handle the start and end of a measurement. I have this change working for me, is this a change you would be interested in taking?

**Comment by jrouwe** [comment]
The current solution tries to keep the overhead of the profiler to a minimum by avoiding a function pointer call. You're right though that this solution does not work if you're compiling Jolt as a dynamic library. Send me a PR with your changes and I'll take a look. I may edit it a bit before integrating.
